src/manage_sql_nvts.c

```diff
@@ -1147,30 +1212,605 @@ manage_complete_nvt_cache_update (GList *nvts_list, GList *nvt_preferences_list)
+
+          if (type == NULL)
+            {
+              GString *debug = g_string_new ("");
+              g_warning ("%s: VT_PARAM missing type attribute", __FUNCTION__);
+              print_entity_to_string (vt_param, debug);
+              g_warning ("%s: VT_PARAM: %s", __FUNCTION__, debug->str);
+              g_string_free (debug, TRUE);
+            }
+          else if (id == NULL)
+            {
+              GString *debug = g_string_new ("");
+              g_warning ("%s: VT_PARAM missing id attribute", __FUNCTION__);
+              print_entity_to_string (vt_param, debug);
+              g_warning ("%s: VT_PARAM: %s", __FUNCTION__, debug->str);
+              g_string_free (debug, TRUE);
+            }
+          else if (name == NULL)
+            {
+              GString *debug = g_string_new ("");
+              g_warning ("%s: VT_PARAM missing NAME", __FUNCTION__);
+              print_entity_to_string (vt_param, debug);
+              g_warning ("%s: VT_PARAM: %s", __FUNCTION__, debug->str);
+              g_string_free (debug, TRUE);
+            }
+          else if (def == NULL)
+            {
+              GString *debug = g_string_new ("");
+              g_warning ("%s: VT_PARAM missing DEFAULT", __FUNCTION__);
+              print_entity_to_string (vt_param, debug);
+              g_warning ("%s: VT_PARAM: %s", __FUNCTION__, debug->str);
+              g_string_free (debug, TRUE);
+            }
+          else
+            {
+              gchar *full_name;
+              preference_t *preference;
+
+              full_name = g_strdup_printf ("%s:%s:%s:%s",
+                                           oid,
+                                           id,
+                                           type,
+                                           entity_text (name));
+
+              blank_control_chars (full_name);
+              preference = g_malloc0 (sizeof (preference_t));
+              preference->name = full_name;
+              preference->value = g_strdup (entity_text (def));
+              *preferences = g_list_prepend (*preferences, preference);
+            }
+        }
+
+      children = next_entities (children);
+    }
+
+  return 0;
+}
+
+/**
+ * @brief Update NVT from VT XML.
+ *
+ * @param[in]  vt           OSP GET_VTS VT element.
+ * @param[in]  preferences  All NVT preferences.
+ *
+ * @return 0 success, -1 error.
+ */
+static int
+update_nvt_from_vt (entity_t vt, GList **preferences)
+{
+  const char *id;
+  entity_t name, vt_refs, custom, family, category;
+  gchar *cve, *bid, *xref, *tag;
+  gchar *cvss_base, *parsed_tags;
+
+  assert (preferences);
+
+  id = entity_attribute (vt, "id");
+  if (id == NULL)
+    {
+      g_warning ("%s: VT missing id attribute", __FUNCTION__);
+      return -1;
+    }
+
+  name = entity_child (vt, "name");
+  if (name == NULL)
+    {
+      g_warning ("%s: VT missing NAME", __FUNCTION__);
+      return -1;
+    }
+
+  vt_refs = entity_child (vt, "vt_refs");
```

Comment:
This was updated, should be
```suggestion
  vt_refs = entity_child (vt, "refs");
```

Reply:
I think we are making bad naming choices:

OSP: VT/VT_REFS/REF  (VT_REFS but contains REF)
libs: nvti_t contains vtref_t.  (no space like in OSP)

What I would expect:
OSP: VT/REFS/REF
libs: gvm_vt_t contains gvm_ref_t

Reply:
Thanks for changing VT_PARAMS.  PR uses REFS and PARAMS now.